src/DBFTPlugin/ConsensusService.cs

```diff
@@ -423,17 +423,39 @@ private void OnPrepareRequestReceived(ExtensiblePayload payload, PrepareRequest
                 Log($"Timestamp incorrect: {message.Timestamp}", LogLevel.Warning);
                 return;
             }
+
             if (message.TransactionHashes.Any(p => NativeContract.Ledger.ContainsTransaction(context.Snapshot, p)))
             {
                 Log($"Invalid request: transaction already exists", LogLevel.Warning);
                 return;
             }
 
+            // Possible vulnerability of the random number, dishonest nodes cooperate to
+            // attck the system. 
+            // TODO: make the VRF seed a few more blocks ahead to prevent view change
+            byte[] nonce;
+            if (message.BlockIndex > 1000)
+                nonce = VRF.Verify(context.Validators[message.ValidatorIndex], message.VRFProof, NativeContract.Ledger.GetBlockHash(context.Snapshot, message.BlockIndex - 1000).ToArray());
+            else
+            {
+                nonce = VRF.Verify(context.Validators[message.ValidatorIndex], message.VRFProof, message.PrevHash.ToArray());
+            }
+
+            if (nonce == null)
+            {
+                Log($"Random number verification failed: {message.VRFProof}", LogLevel.Warning);
```

Comment:
I prefer throw an error in random method instead of break the consensus

Reply:
And maybe ask ChangeView.

Reply:
> And maybe ask ChangeView.

I just follow the way how you process other check failures. If you think ChangeView is better, I can add it here.

Reply:
> I prefer throw an error in random method instead of break the consensus

Updated by adding exception